test/flatten.spec.js

```diff
@@ -0,0 +1,22 @@
+require('../aee')
+const expect = require('chai').expect
+const data = require('./data')
+
+describe('#flatten', () => {
+  it('returns empty array when applied on an empty array', () => {
+    expect([].flatten()).to.be.empty
+  })
+
+  it('returns the same array when no flatten is needed', () => {
+    expect(data.numbers.flatten()).to.eql(data.numbers)
+    expect(data.letters.flatten()).to.eql(data.letters)
+    expect(['a']).to.eql(['a'])
+  })
+
+  it('returns the correct answer', () => {
+    expect(data.unflattenedArrayNumbersS.flatten()).to.eql(data.flattenedArrayNumbersS)
```

Comment:
Please add a test to validate your method is returning a new array and it's not modifying the original one.